app/domain/export/tabular/people/person_row.rb

```diff
@@ -24,9 +24,20 @@ def gender
 
     def roles
       if entry.try(:role_with_layer).present?
-        entry.roles.zip(entry.role_with_layer.split(', ')).map { |arr| arr.join(' ') }.join(', ')
+        entry_roles.zip(entry.role_with_layer.split(', ')).map { |arr| arr.join(' ') }.join(', ')
       else
-        entry.roles.map { |role| "#{role} #{role.group.with_layer.join(' / ')}" }.join(', ')
+        entry_roles.map { |role| "#{role} #{role.group.with_layer.join(' / ')}" }.join(', ')
+      end
+    end
+
+    def entry_roles
+      if restrict_to_roles.present?
+        entry
+          .roles
+          .where(type: restrict_to_roles)
+          .where(group_id: restrict_to_group_ids)
+      else
+        entry.roles
```

Comment:
Ich glaube wir haben einen Überlegungsfehler in der Tech Spec drin. Problemfall zum reproduzieren:
1. Ich habe zwei Gruppen vom Typ Arbeitsgruppe, nennen wir sie Gruppe A und Gruppe B
2. Ich erstelle eine Person mit der Rolle "Leitung" in Gruppe A und den Rollen "Leitung" und "PräsidentIn" in Gruppe B
3. Ich erstelle ein Abo das die Rolle "Leitung" von Gruppe A und die Rolle "PräsidentIn" von Gruppe B einschliesst
4. Ich exportiere die Abonnenten mit der neuen Checkbox aktiviert

**Erwartetes Ergebnis**: Ich sehe die Rolle "Leitung Gruppe A" und die Rolle "PräsidentIn Gruppe B" im Export
**Tatsächliches Ergebnis**: Ich sehe alle drei Rollen "Leitung Gruppe A", "Leitung Gruppe B" und "PräsidentIn Gruppe B", obwohl die Rolle "Leitung Gruppe B" nicht zur Inklusion im Abo beisteuert.

Grund ist wohl dass wir die restrict_to_roles und restrict_to_group_ids separiert übergeben. Die Rolle "Leitung Gruppe B" wird nämlich sowohl von den restrict_to_roles akzeptiert wie auch von den restrict_to_group_ids. Die Kombination "Leitung" und "Gruppe B" ist aber bei den Abo-Subscriptions nicht eingeschlossen. Also müssen wir eher eine Liste von Gruppe-Rolle-Paaren an den Exporter übergeben.

Reply:
ah I see ... also wenn die Rollen gleich heissen und wir zwei verschiedene Gruppen haben.
ja wir brauchen vermutlich wirklich eine role/group Unterscheidung.

ich habe mich auch schon gefragt ob wir nicht einfach nur nach gruppen ids filtern möchten. dann würden immer alle Rollen angezeigt der aktuellen Gruppen. das würde es technisch auch einfacher machen. was meinst du?
ich habe schon sehr viel Zeit für dieses Feature aufgewendet ...